src/clj/fluree/db/reasoner/owl_datalog.cljc

```diff
@@ -1,54 +1,31 @@
 (ns fluree.db.reasoner.owl-datalog
-  (:require [clojure.string :as str]
-            [fluree.db.json-ld.iri :as iri]
+  (:require [fluree.db.json-ld.iri :as iri]
             [fluree.db.constants :as const]
             [fluree.db.util.core :as util :refer [try* catch*]]
             [fluree.db.util.log :as log]))
 
 ;; conversions of owl statements to datalog
 
-(defn unwrap-list
-  "If values are contained in a @list, unwraps them.
-  @list can look like:
-  {ex:someProperty [{@list [ex:val1 ex:val2]}]}
-  or in single-cardinality form:
-  {ex:someProperty {@list [ex:val1 ex:val2]}}"
+(defn only-named-ids
+  "Returns all @id values that are not blank nodes
+  from either an ordered list or a set of objects."
   [vals]
-  (let [first-val (if (sequential? vals)
-                    (first vals)
-                    vals)
-        list-vals (when (map? first-val)
-                    (or (:list first-val)
-                        (get first-val "@list")))]
-    (or list-vals
-        vals)))
-
-(defn get-types
-  [node]
-  (or (:type node)
-      (get node "@type")))
-
-(defn get-all-ids
-  "Returns all @id values from either an ordered list or a set of objects.
-  Filters out any scalar (@value) values and blank nodes."
-  [vals]
-  (try*
-    (->> (util/sequential vals)
-         unwrap-list
-         (into [] (comp
-                    (map util/get-id)
-                    (remove nil?)
-                    (remove iri/blank-node-id?))))
-    (catch* e
-            (throw (ex-info (str "Unexpected error extracting all @id values from owl statement:" vals)
-                            {:status 500
-                             :error  :db/unexpected-error}
-                            e)))))
+  (->> (util/unwrap-list vals)
```

Comment:
I think this reads easier without the `->>` since it's just one step.

Reply:
Addressed in 70b957ff934dc64cc981bb38bfcfe6c943bf4dbc